src/main/java/life/qbic/io/commandline/PostmanCommandLineOptions.java

```diff
@@ -1,85 +1,161 @@
   public String passwordEnvVariable;
 
-  // this consumes all parameters that are not labeled!
-  @Parameters(paramLabel = "SAMPLE_ID", description = "one or more QBiC sample ids")
-  public List<String> ids;
-
-  @Option(
-      names = {"-f", "--file"},
-      description = "a file with line-separated list of QBiC sample ids")
-  public Path filePath;
-
-  @Option(
-      names = {"-c", "--conserve"},
-      description = "Conserve the file path structure during download")
-  public boolean conservePath;
-
   @Option(
-      names = {"--print"},
-      description = "print available datasets for the provided samples")
-  public boolean printDatasets;
+          names = {"-as", "-as_url"},
+          description = "ApplicationServer URL",
+          scope = CommandLine.ScopeType.INHERIT)
+  public String as_url = "https://qbis.qbic.uni-tuebingen.de/openbis/openbis";
 
   @Option(
-      names = {"-b", "--buffer-size"},
-      description =
-          "dataset download performance can be improved by increasing this value with a multiple of 1024 (default)."
-              + " Only change this if you know what you are doing.")
-  public int bufferMultiplier = 1;
+          names = {"-dss", "--dss_url"},
+          description = "DataStoreServer URL",
+          scope = CommandLine.ScopeType.INHERIT)
+  public String dss_url = "https://qbis.qbic.uni-tuebingen.de/datastore_server";
 
   @Option(
-      names = {"-s", "--suffix"},
-      description = "returns all files of datasets containing the supplied suffix")
-  public List<String> suffixes = new ArrayList<>();
+          names = {"-f", "--file"},
+          description = "a file with line-separated list of QBiC sample ids",
+          scope = CommandLine.ScopeType.INHERIT)
+  public Path filePath;
 
   @Option(
-      names = {"-r", "--regex"},
-      description = "returns all files of datasets using your supplied regular expression")
-  public List<String> regexPatterns = new ArrayList<>();
+          names = {"-h", "--help"},
+          usageHelp = true,
+          description = "display a help message and exit",
+          scope = CommandLine.ScopeType.INHERIT)
+  public boolean helpRequested = false;
 
+  // not used currently
   @Option(
       names = {"-t", "--type"},
-      description = "filter for a given openBIS dataset type")
+      description = "filter for a given openBIS dataset type",
+      scope = CommandLine.ScopeType.INHERIT)
   public String datasetType = "";
 
-  @Option(
-      names = {"-dss", "--dss_url"},
-      description = "DataStoreServer URL")
-  public String dss_url = "https://qbis.qbic.uni-tuebingen.de/datastore_server";
+  private List<String> verifyProvidedIdentifiers() throws IOException {
+    if ((ids == null || ids.isEmpty()) && filePath == null) {
+      System.out.println(
+          "You have to provide one ID as command line argument or a file containing IDs.");
+      System.exit(1);
+    } else if ((ids != null) && (filePath != null)) {
```

Comment:
```suggestion
    } else if (nonNull(ids) && nonNull(filePath)) {
```
from `import static java.util.Objects.nonNull;`
As of java 8, the `java.utils.Objects` class provides some helpful methods to deal with objects. IMHO this is more readable.

Reply:
Thank you for this tipp, I haven't heard of this before but it really is easier to read. I'll change it:)